src/components/scene/pool.js

```diff
@@ -0,0 +1,103 @@
+var debug = require('../../utils/debug');
+var register = require('../../core/component').registerComponent;
+
+var warn = debug('components:pool:warn');
+
+/**
+ * Pool component.
+ * It initialized a pool of entities that will be reused.
+ * It avoids creating and destroying the same kind entities
+ * in dynamic scenes. Useful for example in a game where you want
+ * to reuse enemies entities
+ *
+ * @member {array} poolEls - Available entities in the pool.
+ * @member {array} pooledEls - Entities of the pool in use.
+ *
+ */
+module.exports.Component = register('pool', {
+  schema: {
+    mixin: {default: ''},
+    size: {default: 0},
+    dynamicSize: {default: false}
+  },
+
+  multiple: true,
+
+  init: function () {
+    this.createEntities();
+  },
+
+  createEntities: function () {
+    var mixin = this.data.mixin;
+    if (!mixin) { return; }
+    this.poolEls = [];
+    this.pooledEls = [];
+    for (var i = 0; i < this.data.size; ++i) {
+      this.createEntity();
+    }
+  },
+
+  update: function (oldData) {
```

Comment:
Where is that stated? Is not a policy difficult to understand, enforce and context dependent? Sometimes we sort things alphabetically and sometimes we don't. I recently realized that the order of variables or methods in a file does not help me much. On the other hand it introduces friction and sometimes you have to do weird contortions to conform to the policy like for instance when sorting the variables alphabetically.